interface/main/messages/templates/linked_documents.php

```diff
@@ -0,0 +1,237 @@
+ * that will hit the server and validate the xsd and schematron of the CDA.  The errors are then populated and displayed
+ * on the screen with associated error counts being displayed if they are available.  Each document is handled separately
+ * as an async validation process.
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <snielson@discoverandchange.com>
+ * @copyright Copyright (c) 2022 Discover and Change, Inc. <snielson@discoverandchange.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Database\QueryUtils;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Services\Cda\CdaValidateDocumentObject;
+
+// Get the related document IDs if any.
+$linkedDocsSql = "SELECT id1 FROM gprelations WHERE " .
+    "type1 = ? AND type2 = ? AND id2 = ?";
+$tmp = QueryUtils::fetchRecords($linkedDocsSql, ['1','6',$noteid]);
+if (empty($tmp)) {
+    return;
+}
+
+$enc_list = [];
+if (!empty($prow)) {
+    $results = QueryUtils::fetchRecords("SELECT fe.encounter,fe.date,openemr_postcalendar_categories.pc_catname FROM form_encounter AS fe " .
+        " LEFT JOIN openemr_postcalendar_categories ON fe.pc_catid=openemr_postcalendar_categories.pc_catid  WHERE fe.pid = ? ORDER BY fe.date DESC", array($prow['pid']));
+    foreach ($enc_list as $row) {
+        $enc_list[] = [
+            'encounter' => $row['encounter'],
+            'pc_catname' => xl_appt_category($row['pc_catname']),
+            'date' => oeFormatShortDate(date("Y-m-d", strtotime($row['date'])))
+        ];
+    }
+}
+
+// if we have phimail enabled, we are going to find out if our document is an xml file, if so, we are going to do a validation
+// report on the document.  This would be a great thing for caching if we wanted to do that, we'll just compute on the fly for now.
+$records = [];
+$prow = $prow ?? null;
+$cdaDocumentValidator = new CdaValidateDocumentObject();
+foreach ($tmp as $record) {
+    $d = new Document($record['id1']);
+    $docInformation = [
+        'documentId' => $d->get_id(),
+        'title' => $d->get_name() . "-" . $d->get_id(),
+        'isCda' => false,
+        'hasPatient' => false,
+        'requiresValidation' => false
+    ];
+    if (!empty($prow)) {
+        $docInformation['hasPatient'] = true;
+        $docInformation['pname'] = $prow['fname'] . " " . $prow['lname'];
+        $docInformation['pid'] = $prow['pid'];
+        $docInformation['pubpid'] = $prow['pubpid'];
+        $docInformation['DOB'] = $prow['DOB'];
+    }
+    if ($cdaDocumentValidator->isCdaDocument($d)) {
+        $docInformation['isCda'] = true;
+        $docInformation['requiresValidation'] = true;
+    }
+    $records[] = $docInformation;
+}
+$twig = new TwigContainer(null, $GLOBALS['kernel']);
```

Comment:
is this $twig being used in this script? (if not, then could remove from use block also)

Reply:
Moved it up higher.